sim/paladin/holy_shield.go

```diff
@@ -0,0 +1,116 @@
+package paladin
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/wowsims/sod/sim/core"
+	"github.com/wowsims/sod/sim/core/stats"
+)
+
+var HolyShieldValues = []struct {
+	level    int32
+	spellID  int32
+	procID   int32
+	manaCost float64
+	damage   float64
+}{
+	{level: 30, spellID: 20925, procID: 20955, manaCost: 150, damage: 65},
+	{level: 50, spellID: 20927, procID: 20956, manaCost: 195, damage: 95},
+	{level: 60, spellID: 20928, procID: 20957, manaCost: 240, damage: 130},
+}
+
+func (paladin *Paladin) registerHolyShield() {
+
+	if !paladin.Talents.HolyShield {
+		return
+	}
+
+	var numCharges int32 = 4
+	defendersResolveSPAura := core.DefendersResolveSpellDamage(paladin.GetCharacter())
+	blockBonus := 30.0 * core.BlockRatingPerBlockChance
+
+	for i, values := range HolyShieldValues {
+
+		rank := i + 1
+		level := values.level
+		spellID := values.spellID
+		procID := values.procID
+		manaCost := values.manaCost
+		damage := values.damage
+
+		if paladin.Level < level {
+			break
+		}
+
+		holyShieldProc := paladin.RegisterSpell(core.SpellConfig{
+			ActionID:    core.ActionID{SpellID: procID},
+			SpellCode:   SpellCode_PaladinHolyShieldProc,
+			SpellSchool: core.SpellSchoolHoly,
+			DefenseType: core.DefenseTypeMagic,
+			ProcMask:    core.ProcMaskSpellDamage,
+
+			RequiredLevel: int(level),
+			Rank:          rank,
+
+			DamageMultiplier: 1,
+			ThreatMultiplier: 1.2,
+			BonusCoefficient: 0.05,
+
+			ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+				spell.CalcAndDealDamage(sim, target, damage, spell.OutcomeMagicHitAndCrit)
```

Comment:
Can holy shield's damage be resisted?

Reply:
Yes, it can be. (e.g. https://sod.warcraftlogs.com/reports/XPpkgMHq1ZwKDcj2#fight=21&type=damage-done&source=4&ability=456544 )